src/CacheWithResilience.php

```diff
@@ -0,0 +1,165 @@
+use Symfony\Component\Cache\CacheItem;
+
+/**
+ * An implementation of stale-if-error for the Cache library.
+ * This class is compatible with BBC\ProgrammesCachingLibrary\Cache, the only difference when calling the functions
+ * is that you can call getItem() with an extra parameter to get the stale value.
+ */
+class CacheWithResilience extends Cache
+{
+    /** @var LoggerInterface  */
+    private $logger;
+
+    /** @var \Exception[]  */
+    private $whitelistedExceptions;
+
+    // How long items will persist in the cache server.
+    // This also mean for how long we could serve stale content in case of issues
+    private const RESILIENCE_TTL = 21600; // 6h
+
+    public function __construct(
+        LoggerInterface $logger,
+        CacheItemPoolInterface $cachePool,
+        string $prefix,
+        array $cacheTimes = [],
+        array $whitelistedExceptions = []
+    ) {
+        $this->logger = $logger;
+        $this->whitelistedExceptions = $whitelistedExceptions;
+        parent::__construct($cachePool, $prefix, $cacheTimes);
+    }
+
+    /**
+     * This function get the item if exist in memory and the TTL
+     *
+     * @param string $key
+     * @param bool $returnStaleValue
+     * @return CacheItemInterface
+     */
+    public function getItem(string $key, bool $returnStaleValue = false): CacheItemInterface
+    {
+        $key = $this->standardiseKey($key);
+        if ($this->flushCacheItems) {
+            $this->cachePool->deleteItem($key);
+        }
+
+        $item = $this->cachePool->getItem($key);
+        if ($item->isHit()) {
+            $value = $item->get();
+            // we have 2 TTL's, one of them is stored with the item as $value['expiresAt'] and is the time
+            // when we want the item to be consider as expired, the second one is the real TTL that we tell redis.
+            // The second one is only mean to be used in case of issues like DB down and only requires to
+            // set $returnStaleValue to true
+            if (false === $returnStaleValue && new DateTime() > $value['expiresAt']) {
+                return new CacheItem();
+            }
+        }
+
+        // in setItem() value gets wrapped into an array. set the item to its original value
+        $val = $item->get();
+        $item->set($val['value']);
+
+        return $item;
+    }
+
+    /**
+     * @param CacheItemInterface $item
+     * @param mixed $value
+     * @param int|string|DateTimeInterface $ttl
+     *   TTL in seconds, or a constant from CacheInterface, or a DateTime to expire at
+     * @return bool
+     */
+    public function setItem(CacheItemInterface $item, $value, $ttl): bool
+    {
+        // this condition calculates the DateTime when the element should be consider as expire
+        if ($ttl instanceof DateTimeInterface) {
+            $expiresAt = $ttl;
+        } elseif (\in_array($ttl, [0, -1, CacheInterface::NONE, CacheInterface::INDEFINITE, CacheInterface::SHORT], true)) {
```

Comment:
Why is short in this array?

Reply:
For safety, I think if we want to cache something for 60 seconds could be because we are more worry about displaying stale information for that key. Do you think is a good approach or do you think there is no need to be worried?